core/fslock/fslock.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_FSLOCK_HPP
+#define FILECOIN_CORE_FSLOCK_HPP
+
+#include <boost/interprocess/sync/file_lock.hpp>
+#include <string>
+#include "common/outcome.hpp"
+
+namespace fc::fslock {
+  class Locker {
+   public:
+    /**
+     * @brief tries to lock file
+     * @param file_lock_path is path to file that we want to lock
+     * @return lock file
+     */
+    static outcome::result<boost::interprocess::file_lock> lock(
+        const std::string &file_lock_path);
+
+   private:
+    static boost::interprocess::interprocess_mutex mutex;
```

Comment:
IMO, need to unlock the mutex in destructor.

Reply:
This mutex wrapped into scope_lock, after each call the lock function it will be unlocked